### PR TITLE
List pokemons and locations

### PR DESCRIPTION
**Pull request by ikelesoglu**
- List fake Pokemons and fake Locations in Recyclerviews
- Update PokedexScreenState Data class
- Update HomeViewModel and HomeViewModelTest accordingly

### CODE REVIEW

File: domain/src/main/java/com/pokedroid/domain/interactors/RetrieveLocations.kt [hunk 1]
```diff
@@ -7,7 +7,7 @@ import io.reactivex.Observable
 class RetrieveLocations : RetrieveInteractor<Unit, List<Location>> {
 
     override fun retrieveBehaviorStream(params: Unit): Observable<List<Location>> {
-        val fakeData = listOf<Location>()
+        val fakeData = listOf(Location(1,"Ilke's Mansion"), Location(2, "Fabio's Mansion"))
```
[the remark this answers is not in the document]
Comment (ikelesoglu) [reply domain/src/main/java/com/pokedroid/domain/interactors/RetrieveLocations.kt]: mine (agreed on it on slack)

File: presentation/build.gradle [hunk 2]
```diff
@@ -4,11 +4,11 @@ apply plugin: 'kotlin-kapt'
 apply plugin: "de.mannodermaus.android-junit5"
 
 android {
-    compileSdkVersion 28
+    compileSdkVersion 27
```
[the remark this answers is not in the document]
Comment (ikelesoglu) [reply presentation/build.gradle]: fixed

File: presentation/src/main/java/com/pokedroid/presentation/HomeActivity.kt [hunk 3]
```diff
@@ -1,24 +1,71 @@
...
 import com.pokedroid.presentation.dagger.DaggerHomeComponent
 import javax.inject.Inject
 
-class HomeActivity: AppCompatActivity() {
+class HomeActivity : AppCompatActivity() {
 
     @Inject
     lateinit var viewModel: HomeViewModel
 
+    private lateinit var pokemonListView: RecyclerView
```
[the remark this answers is not in the document]
Comment (ikelesoglu) [reply presentation/src/main/java/com/pokedroid/presentation/HomeActivity.kt]: I don't think I just do stuff :D

File: presentation/src/main/java/com/pokedroid/presentation/HomeActivity.kt [hunk 4]
```diff
@@ -1,24 +1,71 @@
...
+        observeLocations()
+    }
+
+    private fun observerPokemons() {
+        pokemonViewManager = LinearLayoutManager(this)
+
+        viewModel.pokedexLiveData.observe(this, Observer<PokedexScreenState?> {
+            when (it) {
+                is PokedexScreenState.Data -> {
+                    pokemonListView = findViewById<RecyclerView>(R.id.pokemon_list).apply {
```
[the remark this answers is not in the document]
Comment (ikelesoglu) [reply presentation/src/main/java/com/pokedroid/presentation/HomeActivity.kt]: lessons learned

File: presentation/src/main/java/com/pokedroid/presentation/HomeActivity.kt [hunk 5]
```diff
@@ -1,24 +1,71 @@
...
+
+    private fun observeLocations() {
+        locationViewManager = LinearLayoutManager(this)
+
+        viewModel.pokedexLiveData.observe(this, Observer<PokedexScreenState?> {
+            when (it) {
+                is PokedexScreenState.Data -> {
+                    locationListView = findViewById<RecyclerView>(R.id.location_list).apply {
+                        setHasFixedSize(true)
+                        layoutManager = locationViewManager
```
[the remark this answers is not in the document]
Comment (ikelesoglu) [reply presentation/src/main/java/com/pokedroid/presentation/HomeActivity.kt]: minor facepalm

File: domain/src/main/java/com/pokedroid/domain/interactors/RetrieveLocations.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (fabiocarballo) [reply domain/src/main/java/com/pokedroid/domain/interactors/RetrieveLocations.kt]: sorry, can't approve like this

[the remark this answers is not in the document]
Comment (ikelesoglu) [reply domain/src/main/java/com/pokedroid/domain/interactors/RetrieveLocations.kt]: ok ok "yours"

**Pull request merged**